hooks/salesforce_hook.py

```diff
@@ -1,66 +1,117 @@
-from airflow.hooks.base_hook import BaseHook
+from airflow.contrib.hooks.salesforce_hook import SalesforceHook
+import copy
 from simple_salesforce import Salesforce
+import pandas as pd
+import time
 
-class SalesforceHook(BaseHook):
-    def __init__(
-            self,
-            conn_id,
-            *args,
-            **kwargs
-    ):
-        """
-        Borrowed from airflow.contrib
-
-        Create new connection to Salesforce
-        and allows you to pull data out of SFDC and save it to a file.
-        You can then use that file with other
-        Airflow operators to move the data into another data source
-        :param conn_id:     the name of the connection that has the parameters
-                            we need to connect to Salesforce.
-                            The conenction shoud be type `http` and include a
-                            user's security token in the `Extras` field.
-
-        .. note::
-            For the HTTP connection type, you can include a
-            JSON structure in the `Extras` field.
-            We need a user's security token to connect to Salesforce.
-            So we define it in the `Extras` field as:
-                `{"security_token":"YOUR_SECRUITY_TOKEN"}`
+class SalesforceYieldingHook(SalesforceHook):
+
+    def __init__( self, conn_id, *args, **kwargs):
+        super(SalesforceYieldingHook, self).__init__(conn_id, *args,**kwargs)
+
+
+    def yield_all(self, query,include_deleted=False, **kwargs):
+        """yield paged results and schema for the `query`. This is a convenience
+        wrapper around `query(...)` and `query_more(...)`.
+        * query -- the SOQL query to send to Salesforce, e.g.
+                   SELECT Id FROM Lead WHERE Email = "waldo@somewhere.com"
+        * include_deleted -- True if the query should include deleted records.
         """
 
-        self.sf = None
-        self.conn_id = conn_id
-        self._args = args
-        self._kwargs = kwargs
+        self.log.info("Querying for all objects")
+        result = self.sf.query(query, include_deleted=include_deleted, **kwargs)
+        object_name = result['records'][0]['attributes']['type']
+        schema = self.describe_object(object_name)
+        all_records = []
 
-        # get the connection parameters
-        self.connection = self.get_connection(conn_id)
-        self.extras = self.connection.extra_dejson
+        while True:
+            yield (result,schema)
+            # fetch next batch if we're not done else break out of loop
+            if not result['done']:
+                result = self.sf.query_more(result['nextRecordsUrl'], identifier_is_url=True, include_deleted=include_deleted)
+            else:
+                break
+        None
 
-    def get_conn(self):
+    def format_results(
+        self,
+        query_results,
+        schema,
+        fmt="csv",
```

Comment:
Does defaulting to csv break backwards compat with what you have done @benjamingregory ?